services/viva/microservice/src/lambdas/syncWorkflow.ts

```diff
@@ -0,0 +1,121 @@
+import log from '../libs/logs';
+import config from '../libs/config';
+import * as dynamoDb from '../libs/dynamoDb';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import { CaseItem } from '../types/caseItem';
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+import { VivaWorkflow } from '../types/vivaWorkflow';
+
+type GetCaseResult = Promise<{ Items: CaseItem[] }>;
+interface GetWorkflowParams {
+  personalNumber: string;
+  workflowId: string | null;
+}
+
+export interface GetWorkflowResult {
+  attributes: VivaWorkflow;
+}
+
+export interface Dependencies {
+  getSubmittedOrProcessingOrOngoingCases: (personalNumber: string) => GetCaseResult;
+  updateCaseDetailsWorkflow: (caseKeys: CaseKeys, newWorkflow: VivaWorkflow) => unknown;
+  syncWorkflowSuccess: (detail: Record<string, unknown>) => unknown;
+  getWorkflow: (payload: GetWorkflowParams) => Promise<GetWorkflowResult>;
+}
+
+interface User {
+  personalNumber: string;
+}
+
+interface LambdaDetails {
+  user: User;
+}
+
+export interface LambdaRequest {
+  detail: LambdaDetails;
+}
+
+interface CaseKeys {
+  PK: string;
+  SK: string;
+}
+
+function getSubmittedOrProcessingOrOngoingCases(personalNumber: string): GetCaseResult {
+  const PK = `USER#${personalNumber}`;
+
+  const queryParams = {
+    TableName: config.cases.tableName,
+    KeyConditionExpression: 'PK = :pk',
+    FilterExpression:
+      '(begins_with(#status.#type, :statusTypeSubmitted) or begins_with(#status.#type, :statusTypeProcessing) or begins_with(#status.#type, :statusTypeOngoing)) and provider = :provider',
+    ExpressionAttributeNames: {
+      '#status': 'status',
+      '#type': 'type',
+    },
+    ExpressionAttributeValues: {
+      ':pk': PK,
+      ':statusTypeSubmitted': 'active:submitted',
+      ':statusTypeProcessing': 'active:processing',
+      ':statusTypeOngoing': 'active:ongoing',
+      ':provider': 'VIVA',
+    },
+  };
+
+  return dynamoDb.call('query', queryParams);
+}
+
+function updateCaseDetailsWorkflow(keys: CaseKeys, newWorkflow: VivaWorkflow) {
+  const updateParams = {
+    TableName: config.cases.tableName,
+    Key: {
+      PK: keys.PK,
+      SK: keys.SK,
+    },
+    UpdateExpression: 'SET details.workflow = :newWorkflow',
+    ExpressionAttributeValues: {
+      ':newWorkflow': newWorkflow,
+    },
+    ReturnValues: 'NONE',
+  };
+
+  return dynamoDb.call('update', updateParams);
+}
+
+export async function syncWorkflow(input: LambdaRequest, dependencies: Dependencies) {
+  const { personalNumber } = input.detail.user;
+
+  const userCases = await dependencies.getSubmittedOrProcessingOrOngoingCases(personalNumber);
+
+  const caseList = userCases.Items;
+
+  const isEmptyCaseList = Array.isArray(caseList) && !caseList.length;
```

Comment:
**Optional**

I think that this is enough:

```js
const isEmptyCaseList = caseList?.length > 0; 
```

This would only fail if caseList is an object with the `length` property being a number.